lib/stores/FileStore.ts

```diff
@@ -188,4 +193,48 @@ export default class FileStore extends DataStore {
 
     this.configstore.set(file_id, file)
   }
+
+  async deleteExpired(): Promise<number> {
+    const now = new Date()
+    const toDelete: Promise<void>[] = []
+    const fileStats = await Promise.all(
+      Object.keys(this.configstore.all).map((file_id) => {
+        return this.getUpload(file_id)
+      })
+    )
+
+    let index = 0
+    for (const file_id of Object.keys(this.configstore.all)) {
```

Comment:
I would recommend against calling this.configstore.all multiple times as content can and will change between calls.
When content of configstore changes, indices will not match anymore and entire loop will fail to do anything meaningful.